README.md

```diff
@@ -1105,6 +1108,89 @@ However, the following rules of thumb are useful to keep in mind:
   Alternatively, the entire OrientDB class and all edges that may point to it may be omitted
   entirely from the GraphQL schema.
 
+## Query Cost Estimation
+
+Some GraphQL queries executed over a database may use a lot of memory and/or have a long execution
+time. In such cases, it's better to warn the user, and suggest they revise their query. So it's
+worthwhile to estimate a query's cost (in terms of time taken and memory used) using statistics
+about the database without executing the query.
+
+To this end, a prototypical version for cost estimation is included in
+`graphql_compiler/cost_estimation/`.
+
+Since the time and memory required for query execution varies much depending on the database the
+query is being executed on, the current version of the cost estimator estimates the query's
+cardinality (i.e. the number of rows in the result) as a rough proxy for time/memory usage.
+
+To get estimates for the number of result rows for a given query and statistics object, use the
+`estimate_query_result_cardinality()` function in
+`graphql_compiler/cost_estimation/cardinality_estimator.py`.
+
+### Example of Query Cost Estimation
+
+```python
+from graphql_compiler.cost_estimation.cardinality_estimator import estimate_query_result_cardinality
+from graphql_compiler.cost_estimation.statistics import LocalStatistics
+
+# The cost estimator needs the schema graph for information about the classes in the schema.
+# Get schema graph from hypothetical Animals database.
+schema_graph = obtain_schema_graph_from_animals_database()
+
+# This is the query and parameters that the estimator will produce a cardinality estimate for.
+graphql_query = '''
+{
+    Animal {
+        name @filter(op_name: "in_collection", value: ["$name_params"])
+        out_Animal_ImportantEvent {
+            ... on FeedingEvent {
+                name @output(out_name: "event_name")
+                uuid @output(out_name: "event_id")
+            }
+        }
+    }
+}
+'''
+params = {'name_params': ['Albert', 'Betty', 'Charlie']}
+
+# This dictionary describes the number of Animal vertices, Animal_ImportantEvent
+# edges, and FeedingEvent vertices.
+class_counts = {
+    'Animal': 6,
+    'Animal_ImportantEvent': 20,
+    'FeedingEvent': 4,
+}
+
+# This dictionary describes the number of Animal_ImportantEvent edges specifically
+# connecting Animal and FeedingEvent vertices.
+vertex_edge_vertex_counts = {
+    ('Animal', 'Animal_ImportantEvent', 'FeedingEvent'): 10,
+}
+
+# We include statistics for class_counts and vertex_edge_vertex_counts.
+statistics_object = LocalStatistics(class_counts,
+                                    vertex_edge_vertex_counts=vertex_edge_vertex_counts)
+
+print(estimate_query_result_cardinality(schema_graph, statistics_object, graphql_query, params))
+# The estimator expects approximately 5.0 rows in the result.
```

Comment:
If you don't explain the logic of how you got to the result, then saying that "The estimator expects approximately 5.0 rows in the result." doesn't convey any useful information to the reader.